MapleServer2/PacketHandlers/Game/BeautyHandler.cs

```diff
@@ -360,7 +360,9 @@ private static void HandleTeleport(GameSession session, PacketReader packet)
                     return;
             }
 
-            MoveFieldHandler.HandleInstanceMove(session, (int) mapId);
+            session.Player.ReturnCoord = session.FieldPlayer.Coord;
+            session.Player.ReturnMapId = session.Player.MapId;
+            session.Player.Warp((int) mapId, instanceId: session.Player.CharacterId);
```

Comment:
this is confusing, use one format or the other, `mapId: (int) mapId, instanceId: session.Player.CharacterId`